dwh/indicator_lib.py

```diff
@@ -82,14 +82,48 @@ def calc_TX_PVLS(patient_agg_obs: pandas.DataFrame, VL_code: str,
   if end_date_str:
     end_date = datetime.strptime(end_date_str, '%Y-%m-%d')
   temp_df = patient_agg_obs[(patient_agg_obs['code'] == VL_code)].copy()
+  # separate value from date
+  temp_df[['vl_max_date', 'latest_vl_value']] = temp_df.max_date_value.str.split('_SeP_', expand=True)
+  temp_df['latest_vl_value'] = temp_df['latest_vl_value'].astype(float)
   # Note the above copy is used to avoid setting a new column on a slice next:
-  temp_df['sup_VL'] = (temp_df['max_value'] < failure_threshold)
+  temp_df['sup_VL'] = (temp_df['latest_vl_value'] < failure_threshold)
   temp_df['buckets'] = temp_df.apply(
       lambda x: _agg_buckets(x.birthDate, x.gender, datetime.today()), axis=1)
   temp_df_exp = temp_df.explode('buckets')
   temp_df_exp = temp_df_exp.groupby(['sup_VL', 'buckets'], as_index=False)\
       .count()[['sup_VL', 'buckets', 'patientId']]\
-      .rename(columns={'patientId': 'count'})
+      .rename(columns={'patientId': 'sup_VL_count'})
   # calculate ratio
-  temp_df_exp['ratio'] = temp_df_exp['count']/temp_df['sup_VL'].count()
+  temp_df_exp['sup_VL_ratio'] = temp_df_exp['sup_VL_count']/temp_df['sup_VL'].count()
   return temp_df_exp
+
+
+def calc_TX_NEW(patient_agg_obs:  pandas.DataFrame, ARV_plan: str, start_drug: List,
+    end_date_str: str = None) -> pandas.DataFrame:
+  """Calculates TX_NEW indicator with its corresponding disaggregations.
```

Comment:
Please leave a blank line after the one-line summary ([style-guide](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings)).

Reply:
noted